src/tool/subcommands/benchmark_cmd.rs

```diff
@@ -110,6 +122,34 @@ async fn benchmark_car_streaming(input: Vec<PathBuf>) -> Result<()> {
     Ok(())
 }
 
+// Concatenate a set of CAR files and measure how quickly we can stream the
+// blocks, while inspecting them. This a benchmark we could use for setting
+// realistic expectations in terms of DFS graph travels, for example.
+async fn benchmark_car_streaming_inspect(input: Vec<PathBuf>) -> Result<()> {
+    let mut sink = indicatif_sink("traversed");
+    let ipld_to_cid = |ipld| {
+        if let Ipld::Link(cid) = ipld {
+            return Some((cid, ()));
+        }
+        None
+    };
+
+    let mut s = Box::pin(
+        futures::stream::iter(input)
+            .then(File::open)
+            .map_ok(BufReader::new)
+            .and_then(CarStream::new)
+            .try_flatten(),
+    );
+    while let Some(block) = s.try_next().await? {
+        let block: Block = block;
+        let ipld: Ipld = from_slice_with_fallback(&block.data).unwrap_or(Ipld::Null);
```

Comment:
You can check if the data is DAG_CBOR before parsing it. Then, if it fails to parse, you can propagate the error.

`if cid.codec() == DAG_CBOR { ... }`